BraintreeCore/src/main/java/com/braintreepayments/api/Configuration.kt

```diff
@@ -0,0 +1,410 @@
+package com.braintreepayments.api
+
+import androidx.annotation.RestrictTo
+import org.json.JSONException
+import org.json.JSONObject
+
+/**
+ * Contains the remote configuration for the Braintree Android SDK.
+ */
+// NEXT MAJOR VERSION: remove 'open' modifiers, Java classes/methods are open by default
+// Ref: https://kotlin-quick-reference.com/102c-R-open-final-classes.html
+open class Configuration internal constructor(configurationString: String?) {
```

Comment:
I'm confused why we need to mark this as `open` when converting to Kotin? In [java](https://github.com/braintree/braintree_android/blob/4ffc342d2dd2386f5c1628fa2e599faa88d0573e/BraintreeCore/src/main/java/com/braintreepayments/api/Configuration.java#L18), it's currently `public`.

Reply:
In Kotlin everything defaults to the Java equivalent of `final`. We technically need to mark public facing types as `open` to prevent introducing a breaking change.

Reply:
Got it. Kotlin's default is public / final.